src/main/java/com/mparticle/kits/KochavaKit.java

```diff
@@ -4,44 +4,65 @@
 import android.location.Location;
 
-import com.kochava.android.tracker.Feature;
-import com.kochava.android.tracker.ReferralCapture;
+import com.kochava.base.AttributionUpdateListener;
+import com.kochava.base.DeepLinkListener;
+import com.kochava.base.ReferralReceiver;
+import com.kochava.base.Tracker;
+import com.kochava.base.Tracker.Configuration;
+import com.mparticle.AttributionError;
+import com.mparticle.AttributionResult;
 import com.mparticle.MParticle;
+import com.mparticle.internal.Logger;
+import com.mparticle.internal.MPUtility;
+
+import org.json.JSONException;
+import org.json.JSONObject;
 
 import java.util.HashMap;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 
 public class KochavaKit extends KitIntegration implements KitIntegration.AttributeListener {
+    public static final String ATTRIBUTION_PARAMETERS = "attribution";
+    public static final String DEEPLINK_PARAMETERS = "deeplink";
 
     private static final String APP_ID = "appId";
     private static final String USE_CUSTOMER_ID = "useCustomerId";
     private static final String INCLUDE_ALL_IDS = "passAllOtherIdentities";
     private static final String LIMIT_ADD_TRACKING = "limitAdTracking";
     private static final String RETRIEVE_ATT_DATA = "retrieveAttributionData";
     private static final String ENABLE_LOGGING = "enableLogging";
-    private Feature feature;
 
     private static Map<String, String> identityLink;
 
-    @Override
-    public Object getInstance() {
-        return feature;
-    }
-
     @Override
     public String getName() {
         return "Kochava";
     }
 
     @Override
     protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
-        createKochava();
-        if (feature != null) {
-            feature.setAppLimitTracking(Boolean.parseBoolean(getSettings().get(LIMIT_ADD_TRACKING)));
-            Feature.enableDebug(Boolean.parseBoolean(getSettings().get(ENABLE_LOGGING)));
+        int logLevel = Tracker.LOG_LEVEL_NONE;
+        if (Boolean.parseBoolean(getSettings().get(ENABLE_LOGGING))) {
+            logLevel = Tracker.LOG_LEVEL_DEBUG;
+        }
+        AttributionUpdateListener attributionListener = null;
+        if (Boolean.parseBoolean(getSettings().get(RETRIEVE_ATT_DATA))) {
+            attributionListener = mAttributionListener;
+        }
+
+        Configuration configuration = new Configuration(context.getApplicationContext())
+                .setLogLevel(logLevel)
+                .setAppGuid(getSettings().get(APP_ID))
+                .setAppLimitAdTracking(Boolean.parseBoolean(getSettings().get(LIMIT_ADD_TRACKING)))
+                .setAttributionUpdateListener(attributionListener);
+
+        if (identityLink != null) {
+            configuration.setIdentityLink(new Tracker.IdentityLink().add(identityLink));
         }
+        Tracker.configure(configuration);
+        Tracker.setDeepLinkListener(null, mDeepLinkListener);
```

Comment:
I had forgotten that we actually capture the launch URI which is meant to be passed here and surface it via `MParticle.getInstance().getAppStateManager().getLaunchUri()`. That was need for Button. If/when that interface is changed maybe we should surface it somewhere else. For now - maybe a good idea to pass it in here, otherwise this method might be broken.

Reply:
oh man, good call. I did not realize we retained that